### ISSUE TITLE
em formatting issues using underscore (_)

### ISSUE THREAD

**Issue by oliversturm** [markedjs/marked#1390]
Delimiting underscores are not detected correctly if they are followed by commas. Here's a snippet that demonstrates the issue:

```markdown
_This should be em_, but the end is not detected correctly due to the comma.

_If I have some em_, followed by a comma, and then _some further em_ that's detected correctly, the two are combined.
```

The result of passing this text through marked 0.5.2 is this:

```html
<p>_This should be em_, but the end is not detected correctly due to the comma.</p>
<p><em>If I have some em_, followed by a comma, and then _some further em</em> that’s detected correctly, the two are combined.</p>
```

As you can see, several underscores are still included in the HTML output, and the two "em" parts in the second paragraph have been combined incorrectly.

The  [CommonMark Demo](https://spec.commonmark.org/dingus/) formats the output as expected, leaving no underscores in the HTML.

**Comment by UziTech** [comment]
Thanks for reporting this.

[CommonMark demo](https://spec.commonmark.org/dingus/?text=_This%20should%20be%20em_%2C%20but%20the%20end%20is%20not%20detected%20correctly%20due%20to%20the%20comma.%0A%0A_If%20I%20have%20some%20em_%2C%20followed%20by%20a%20comma%2C%20and%20then%20_some%20further%20em_%20that%27s%20detected%20correctly%2C%20the%20two%20are%20combined.)

[marked demo](https://marked.js.org/demo/?outputType=html&text=_This%20should%20be%20em_%2C%20but%20the%20end%20is%20not%20detected%20correctly%20due%20to%20the%20comma.%0A%0A_If%20I%20have%20some%20em_%2C%20followed%20by%20a%20comma%2C%20and%20then%20_some%20further%20em_%20that%27s%20detected%20correctly%2C%20the%20two%20are%20combined.&options=%7B%0A%20%22baseUrl%22%3A%20null%2C%0A%20%22breaks%22%3A%20false%2C%0A%20%22gfm%22%3A%20true%2C%0A%20%22headerIds%22%3A%20true%2C%0A%20%22headerPrefix%22%3A%20%22%22%2C%0A%20%22highlight%22%3A%20null%2C%0A%20%22langPrefix%22%3A%20%22language-%22%2C%0A%20%22mangle%22%3A%20true%2C%0A%20%22pedantic%22%3A%20false%2C%0A%20%22sanitize%22%3A%20false%2C%0A%20%22sanitizer%22%3A%20null%2C%0A%20%22silent%22%3A%20false%2C%0A%20%22smartLists%22%3A%20false%2C%0A%20%22smartypants%22%3A%20false%2C%0A%20%22tables%22%3A%20true%2C%0A%20%22xhtml%22%3A%20false%0A%7D)

**Comment by styfle** [comment]
I believe this is fixed on master. Maybe we should release soon?

**Comment by oliversturm** [comment]
@UziTech I see you copied some results into your comment. Just to keep things very clear here - I think you made a mistake - the CommonMark output I see is this (note the second line is different from yours):

```html
<p><em>This should be em</em>, but the end is not detected correctly due to the comma.</p>
<p><em>If I have some em</em>, followed by a comma, and then <em>some further em</em> that's detected correctly, the two are combined.</p>
```

**Comment by UziTech** [comment]
looks like this was fixed in 0.6.0

**Comment by oliversturm** [comment]
Confirmed: my original test cases render correctly now. Thanks for the update!
